examples/48-drawindirect/drawindirect.cpp

```diff
@@ -0,0 +1,381 @@
+	
+	// Tetrahedron Model (offset = 8)
+	{ 1.0f,  1.0f,  1.0f, 0xff0000ff },
+	{ 1.0f, -1.0f, -1.0f, 0xff000000 },
+	{-1.0f,  1.0f, -1.0f, 0xff00ff00 },
+	{-1.0f, -1.0f,  1.0f, 0xff00ffff },
+};
+
+static const uint16_t s_multiMeshIndices[48] =
+{
+	// Cube Indicies
+	0, 1, 2, // 0
+	1, 3, 2,
+	4, 6, 5, // 2
+	5, 6, 7,
+	0, 2, 4, // 4
+	4, 2, 6,
+	1, 5, 3, // 6
+	5, 7, 3,
+	0, 4, 1, // 8
+	4, 5, 1,
+	2, 3, 6, // 10
+	6, 3, 7,
+	
+	// Tetrahedron Indices (offset = 36)
+	0, 2, 1,
+	1, 2, 3,
+	0, 3, 2,
+	1, 3, 0,
+};
+
+
+class DrawIndirect : public entry::AppI
+{
+public:
+	DrawIndirect(const char* _name, const char* _description, const char* _url)
+		: entry::AppI(_name, _description, _url)
+	{
+	}
+
+	void init(int32_t _argc, const char* const* _argv, uint32_t _width, uint32_t _height) override
+	{
+		Args args(_argc, _argv);
+
+		m_width  = _width;
+		m_height = _height;
+		m_debug  = BGFX_DEBUG_TEXT;
+		m_reset  = BGFX_RESET_VSYNC;
+		m_sideSize         = 11;
+		m_nDrawElements = 121;
+
+		bgfx::Init init;
+		init.type     = args.m_type;
+		init.vendorId = args.m_pciId;
+		init.platformData.nwh  = entry::getNativeWindowHandle(entry::kDefaultWindowHandle);
+		init.platformData.ndt  = entry::getNativeDisplayHandle();
+		init.resolution.width  = m_width;
+		init.resolution.height = m_height;
+		init.resolution.reset  = m_reset;
+		bgfx::init(init);
+
+		// Enable debug text.
+		bgfx::setDebug(m_debug);
+
+		// Set view 0 clear state.
+		bgfx::setViewClear(0
+			, BGFX_CLEAR_COLOR|BGFX_CLEAR_DEPTH
+			, 0x303030ff
+			, 1.0f
+			, 0
+			);
+
+		// Create vertex stream declaration.
+		PosColorVertex::init();
+		ObjectInstance::init();
+		RenderInstance::init();
+
+		// Create static vertex buffer.
+		m_vbh = bgfx::createVertexBuffer(
+					  bgfx::makeRef(s_multiMeshVertices, sizeof(s_multiMeshVertices) )
+					, PosColorVertex::ms_layout
+					);
+
+		// Create static index buffer.
+		m_ibh = bgfx::createIndexBuffer(
+					bgfx::makeRef(s_multiMeshIndices, sizeof(s_multiMeshIndices) )
+					);
+
+		// Create program from shaders.
+		m_program = loadProgram("vs_instancing", "fs_instancing"); // These are reused from 05-instancing
+		
+		m_indirect_program = BGFX_INVALID_HANDLE;
+		m_indirect_buffer_handle = BGFX_INVALID_HANDLE;
+		m_object_list_buffer = BGFX_INVALID_HANDLE;
+		
+		u_drawParams = bgfx::createUniform("u_drawParams", bgfx::UniformType::Vec4);
+		
+		const bool computeSupported = !!(BGFX_CAPS_DRAW_INDIRECT & bgfx::getCaps()->supported);
+		const bool instancingSupported = !!(BGFX_CAPS_INSTANCING & bgfx::getCaps()->supported);
+		
+		if (computeSupported && instancingSupported) {
+			// Set up indirect program
+			// This is a barebones program that populates the indirect buffer handle with draw requests
+			m_indirect_program = bgfx::createProgram(loadShader("cs_drawindirect"), true);
+			m_indirect_buffer_handle = bgfx::createIndirectBuffer(m_nDrawElements);
+			
+			ObjectInstance objs[m_nDrawElements];
```

Comment:
It would be great that this is also in compute, esp. if you hardcode these values.

Also it's ok to stay here, just if it's here you should somehow correlate it with s_multiMeshIndices, and other data. So that change in data forces change on this location too.